### PR TITLE
PR196 with Jenkins

### PR DESCRIPTION
**Pull request by ed-xmos**
Supersedes https://github.com/xmos/lib_mic_array/pull/196

### CODE REVIEW

**Review by mbanth — APPROVED** [review]
I made a few notes and minor suggestions for improvement, but nothing significant enough to withhold approval.

File: .github/scripts/run_test_apps.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+# Unit tests
+xrun --xscope tests/unit/tests-unit.xe
+
+# Signal/Decimator tests
+pytest ../tests/signal/TwoStageDecimator/ -vv
+
+# Filter design tests
+# TODO fix
+# pytest ../tests/signal/FilterDesign/ -vv
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply .github/scripts/run_test_apps.sh:15]: I have added a comment to the file too https://github.com/xmos/lib_mic_array/issues/201

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            }
+            post {
+                cleanup {
+                    xcoreCleanSandbox()
+                }
+            }
+        }
+        stage('HW tests') {
+            agent {
+                label 'xvf3800'
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply Jenkinsfile:66]: We have plenty of them! I have changed to xcore.ai to make it more obvious

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,118 @@
+@Library('xmos_jenkins_shared_library@v0.25.0') _
+getApproval()
+pipeline {
+    agent none
+    options {
+        disableConcurrentBuilds()
+        skipDefaultCheckout()
+        timestamps()
+        // on develop discard builds after a certain number else keep forever
+        buildDiscarder(logRotator(
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply Jenkinsfile:10]: I will address this in https://github.com/xmos/lib_mic_array/pull/199

**Pull request merged**